File: February10php/BasePlusCommissionEmployee.php
<?php

class BasePlusCommissionEmployee extends CommissionEmployee{
    private float $baseSalary;
    //constructors
    function __construct(string $firstName, string $lastName, string $socialSecurityNumber, float $grossSales, float $commissionRate, float $baseSalary)
    {
        parent::__construct($firstName, $lastName, $socialSecurityNumber, $grossSales, $commissionRate);
        $this->baseSalary = $baseSalary;
    }
    //getter and setter
    public function getBaseSalary(): float{
        return $this->baseSalary;
    }
    public function setBaseSalary(float $newSalary): void{
        if ($newSalary<0){
            throw new InvalidArgumentException("Salary can't be negative");
        }
        $this->baseSalary = $newSalary;
    }
    function earnings(): float
    {
      return  $this->getBaseSalary() + ($this->getGrossSales()*$this->getCommissionRate());
    }
    public function display()
    {
        parent::display()."\nBase Salary is $".$this->baseSalary;
    }


}

$Ama = new BasePlusCommissionEmployee("Ama",
    "Stat",
    "547-222",
    200.00,
    0.47,
    2500.00);
$Ama->earnings();
$Ama->setBaseSalary(5900.00);
print "Ama's new earnings are $".$Ama->earnings();

File: BasePlusCommissionEmployee.java
package org.example;

public class BasePlusCommissionEmployee extends CommissionEmployee{
    public BasePlusCommissionEmployee(String firstName, String lastName, String socialSecurity, double grossSales, double commissionRate, double baseSalary) {
        super(firstName, lastName, socialSecurity, grossSales, commissionRate);
        setBaseSalary(baseSalary);
    }
    private double baseSalary ;

    //methods
    public double getBaseSalary() {
        return baseSalary;
    }
    public void setBaseSalary(double baseSalary) {
        if (baseSalary < 0.0) {
            throw new IllegalArgumentException ("BaseSalary cannot be negative");
        }
        this.baseSalary = baseSalary;
    }

    @Override
    public double earnings(double grossSales, double commissionRate) {
        return getBaseSalary() + super.earnings(grossSales, commissionRate);
    }
    @Override
    public String toString() {
        return "BasePlusCommissionEmployee" + "[baseSalary=]" + baseSalary + "[Employee earnings]" + earnings(getGrossSales(), getCommissionRate());
    }


}


